anya/roman_numeral.rb

```diff
@@ -0,0 +1,36 @@
+class RomanNumeral
+  ROMAN_NUM = {
+              'M' => 1000,
+              'D' => 500,
+              'C' => 100,
+              'L' => 50,
+              'X' => 10,
+              'V' => 5,
+              'I' => 1
+            }
+  ROMAN_NUM.default = 0
+
+  def initialize(number)
+    @number = number.respond_to?(:upcase) ? number.upcase : ""
+    valid?
```

Comment:
we thought it would be strange to have the validate method handle upcasing, since the initializer sets up the variables to be acted upon.